File: pyvulkan/21 - Textures/finished/single_time_commands.py
from config import *

def start_job(commandBuffer):

    vkResetCommandBuffer(
        commandBuffer = commandBuffer,
        flags = 0
    )

    beginInfo = VkCommandBufferBeginInfo(
        flags = VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT
    )
    vkBeginCommandBuffer(
        commandBuffer = commandBuffer, pBeginInfo = beginInfo
    )

def end_job(commandBuffer, queue):

    vkEndCommandBuffer(commandBuffer = commandBuffer)

    submitInfo = VkSubmitInfo(
        commandBufferCount = 1, pCommandBuffers = [commandBuffer,]
    )
    vkQueueSubmit(
        queue = queue, submitCount = 1, pSubmits = [submitInfo,],
        fence = None
    )

    vkQueueWaitIdle(queue = queue)

File: vulkan/Compute Shader Raytracing/03 - Many Spheres/1 host visible buffer/view/vkUtil/single_time_commands.cpp
#include "single_time_commands.h"

void vkUtil::startJob(vk::CommandBuffer commandBuffer) {

	commandBuffer.reset();

	vk::CommandBufferBeginInfo beginInfo;
	beginInfo.flags = vk::CommandBufferUsageFlagBits::eOneTimeSubmit;
	commandBuffer.begin(beginInfo);
}

void vkUtil::endJob(vk::CommandBuffer commandBuffer, vk::Queue submissionQueue) {

	commandBuffer.end();

	vk::SubmitInfo submitInfo;
	submitInfo.commandBufferCount = 1;
	submitInfo.pCommandBuffers = &commandBuffer;
	submissionQueue.submit(1, &submitInfo, nullptr);
	submissionQueue.waitIdle();
}